### ISSUE TITLE
mAP always zero when training is resumed from a particular epoch

### ISSUE THREAD

**Issue by AbhishekHollaAB** [ultralytics/ultralytics#18707]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I had trained a model and had stopped it after a certain number of epochs. Then, to continue training, i used resume=True. The training gets resumed, but after validation, the mask Precision, Recall and both the mAP values are zero.

![Image](https://github.com/user-attachments/assets/1371334c-9738-43df-901b-5ccb1086263b)

### Environment

Ultralytics 8.3.61  Python-3.9.18 torch-1.13.1+cu117 CUDA:0 (Quadro P6000, 24444MiB)
Setup complete ✅ (56 CPUs, 62.8 GB RAM, 116.6/468.4 GB disk)

OS                  Linux-5.4.0-150-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.9.18
Install             pip
RAM                 62.81 GB
Disk                116.6/468.4 GB
CPU                 Intel Xeon E5-2680 v4 2.40GHz
CPU count           56
GPU                 Quadro P6000, 24444MiB
GPU count           1
CUDA                11.7

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.4.0>=1.8.0
torch               ✅ 2.4.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.14.1+cu116>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.3>=2.0.0

### Minimal Reproducible Example

```
from ultralytics import YOLO

# Load a model
# model = YOLO("yolo11x-seg.pt")  # load a pretrained model (recommended for training)
model = YOLO('/home/dcil/Desktop/VyshakB/RWG_Cracks/Data/runs/segment/Jan10Training/weights/best (copy).pt')

# model.train(data="rwg_crack_3/data.yaml", epochs=500, patience=50, save=True, imgsz=1052, 
#             device=0, save_period=30, batch=1, verbose=True, scale=0, name='Jan10Training', optimizer='Adam', lr0=0.01)
model.train(resume=True, data="rwg_crack_3/data.yaml", epochs=500, patience=50, save=True, imgsz=1052, 
            device=0, save_period=30, batch=1, verbose=True, scale=0, name='Jan10Training', optimizer='Adam', lr0=0.01)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AbhishekHollaAB, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be addressed.

If this is a 🐛 Bug Report, please provide a complete [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can debug your issue effectively. Based on the description, the setup and code provided look thorough, but including a simplified and standalone MRE would be very helpful to identify the root cause.

## Debug Recommendations

1. Ensure you are running the latest `ultralytics` version to confirm the issue hasn't been resolved in recent updates. Update with:
   ```bash
   pip install -U ultralytics
   ```
2. Check whether the model weights and training logs prior to resuming are intact and not corrupted to ensure resumption behaves as expected.

3. Consider testing on a smaller and simplified dataset to verify if the issue persists. This can help isolate whether the problem is specific to your dataset or configuration.

## Environments and Options

YOLO can be tested and trained in several verified environments, such as:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For real-time support, consider joining our [Discord community](https://discord.com/invite/ultralytics) 🎧. You can also participate in deeper discussions on our [Discourse platform](https://community.ultralytics.com) or engage with the [Ultralytics Subreddit](https://reddit.com/r/Ultralytics).

## Status and CI

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. Our CI system verifies the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to ensure you get assistance as soon as possible 😊. An Ultralytics engineer will follow up with you shortly for further clarification.

**Comment by Y-T-G** [comment]
Is it the same dataset?

**Comment by AbhishekHollaAB** [comment]
> Is it the same dataset?

Yeah, it is.

**Comment by Y-T-G** [comment]
Can yoh reproduce this with a smaller dataset like `coco128-seg.yaml`?

**Comment by vupcln** [comment]
Too, i think the train pipeline has problem. you can use ultralytics library

**Comment by Y-T-G** [comment]
What's the code you used to reproduce on `coco128-seg.yaml`?

**Comment by AbhishekHollaAB** [comment]
I have not done it on coco128, but the same issue had come on another set of images only while resuming. The issue is with resume=True and not the dataset

**Comment by Y-T-G** [comment]
You can try on coco128-seg.yaml

The issue needs to be reproduced on one of the datasets that are part of Ultralytics. Otherwise, it's hard to say what's wrong because it's not reproducible.

**Comment by AbhishekHollaAB** [comment]
@glenn-jocher Could you please look into this?

**Comment by Y-T-G** [comment]
You will have to provide a minimum reproducible example as mentioned.

https://docs.ultralytics.com/help/minimum_reproducible_example/

Without that, it's hard to debug.
